openstates/de/bills.py

```diff
@@ -53,52 +76,51 @@ def scrape_bill(self, row, chamber, session):
+        # TODO: Is there a way get additional sponsors and cosponsors, and versions/fns via API?
         html_url = 'https://legis.delaware.gov/BillDetail?LegislationId={}'.format(row['LegislationId'])
-        bill.add_source(html_url, mimetype='text/html')
+        bill.add_source(html_url, note='text/html')
 
         html = self.lxmlize(html_url)
 
-        #Additional Sponsors: '//label[text()="Additional Sponsor(s):"]/following-sibling::div/a'
-        additional_sponsors = html.xpath('//label[text()="Additional Sponsor(s):"]/following-sibling::div/a/@href')
+        # Additional Sponsors: '//label[text()="Additional Sponsor(s):"]/following-sibling::div/a'
+        additional_sponsors = html.xpath('//label[text()="Additional Sponsor(s):"]'
+                                         '/following-sibling::div/a/@href')
         for sponsor_url in additional_sponsors:
-            sponsor_id = sponsor_url.replace('https://legis.delaware.gov/LegislatorDetail?personId=', '')
+            sponsor_id = sponsor_url.replace('https://legis.delaware.gov/LegislatorDetail?'
+                                             'personId=', '')
             self.add_sponsor_by_legislator_id(bill, sponsor_id, 'primary')
 
-        #CoSponsors: '//label[text()="Co-Sponsor(s):"]/following-sibling::div/a'
-        cosponsors = html.xpath('//label[text()="Additional Sponsor(s):"]/following-sibling::div/a/@href')
+        # CoSponsors: '//label[text()="Co-Sponsor(s):"]/following-sibling::div/a'
+        cosponsors = html.xpath('//label[text()="Additional Sponsor(s):"]/'
+                                'following-sibling::div/a/@href')
         for sponsor_url in cosponsors:
-            sponsor_id = sponsor_url.replace('https://legis.delaware.gov/LegislatorDetail?personId=','')
+            sponsor_id = sponsor_url.replace('https://legis.delaware.gov/LegislatorDetail?'
+                                             'personId=', '')
             self.add_sponsor_by_legislator_id(bill, sponsor_id, 'cosponsor')
 
         versions = html.xpath('//label[text()="Original Text:"]/following-sibling::div/a/@href')
-        for version in versions:
-            mimetype = self.mime_from_link(version)
-            name = 'Bill Text'
-            bill.add_version(name=name, url=version, mimetype=mimetype)
+        for version_url in versions:
+            media_type = self.mime_from_link(version_url)
+            version_name = 'Bill Text'
+            # on_duplicate='error'
+            bill.add_version_link(version_name, version_url, media_type=media_type)
 
         fiscals = html.xpath('//div[contains(@class,"fiscalNote")]/a/@href')
         for fiscal in fiscals:
             self.scrape_fiscal_note(bill, fiscal)
 
         self.scrape_actions(bill, row['LegislationId'])
-        self.scrape_votes(bill, row['LegislationId'])
-
-        self.save_bill(bill)
+        self.scrape_votes(bill, row['LegislationId'], session)
```

Comment:
It looks like both `scrape_vote` and `scrape_votes` are generators. When I ran this branch, no votes were scraped.